### PR TITLE
feat(ca): add pre-login and more logs

### PR DESCRIPTION
**Pull request by jor-row**
# COMPLETES
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-461413
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-461022

## This pull request addresses

- add logs for submitting call diagnostic events
- adds clientmetrics-prelogin request
- adds event dictionary types from npm

## by making the following changes

- add logs

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios where tested
- manual test
### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts [hunk 1]
```diff
@@ -633,6 +702,14 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
     payload?: ClientEventPayload;
     options?: SubmitClientEventOptions;
   }): Promise<any> {
+    this.logger.log(
+      CALL_DIAGNOSTIC_LOG_IDENTIFIER,
+      'CallDiagnosticMetrics: @buildClientEventFetchRequestOptions. Building request options object for fetch()...',
```
Comment (shnaaz) [inline packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:707]: why ...?

Comment (jor-row) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:707]: I'm guessing there was an issue that Catalin was seeing with a lack of logs. I trust there was a good reason but there is no Jira that describes why the logs were added.

Comment (shnaaz) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:707]: No I didn't mean why log is added, I meant why there are three dots?

Comment (jor-row) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:707]: There were other logs that Catalin added ending in "..." so I followed that "convention"

**Review by mkesavan13 — APPROVED** [review]
Looks good to me

**Pull request merged**